### PR TITLE
Add migration guide

### PR DESCRIPTION
**Pull request by ghost**
This is a work-in-progress. Pushing to get some feedback!

Currently based on #3, will rebase after merge.

### CODE REVIEW

**Comment by ghost** [comment]
Remember to cover `_init` to `new` and removal of `get_`.

**Comment by ghost** [comment]
Should be ready!

File: src/migrating-0-8.md [hunk 1]
```diff
@@ -0,0 +1,418 @@
...
+
+In 0.8, you would write this:
+
+```rust
+#[derive(NativeClass)]
+#[inherit(Object)]
+pub struct Thing;
+
+impl Thing {
+    fn _init(_owner: Object) {}
```
Comment (halzy) [inline src/migrating-0-8.md]: Should there be return types on`_init ()` and`new()`?

File: src/migrating-0-8.md [hunk 2]
```diff
@@ -0,0 +1,418 @@
...
+In 0.9, this should now be written as:
+
+```rust
+use gdnative::prelude::*;
+use gdnative::api::object::ConnectFlags;
+
+object.connect(
+    "foo".into(),
+    owner.to_object(),
+    "_handle_foo".into(),
```
Comment (karroffel) [inline src/migrating-0-8.md]: Aren't those 3 conversions no longer necessary?

File: src/migrating-0-8.md [hunk 3]
```diff
@@ -0,0 +1,418 @@
...
+    T: GodotObject + SubClass<Object>,
+    Access: ThreadAccess,
+{
+    obj.cast()
+}
+```
+
+Note that this function is also polymorphic over the `Access` typestate, which is explained the the following section.
+
+## Typestates
```
Comment (halzy) [inline src/migrating-0-8.md:331]: I really liked this section, it resonated with me more than the docs on`Ref`

**Review by halzy — APPROVED** [review]
Twas a good read.

**Comment by ghost** [comment]
Good catch! Updated the example code.

**Pull request merged**